corehq/apps/settings/tests/test_views.py

```diff
@@ -105,29 +107,39 @@ def _create_view_for_user(self, user):
 
 
 @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=True)
-class TwoFactorSetupView_FormKwargs_Tests(SimpleTestCase):
+class TwoFactorSetupView_RegisteredMethods_Tests(SimpleTestCase):
+
     @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=False)
-    def test_phone_methods_are_prohibited_when_settings_are_disabled(self):
+    def test_phone_methods_are_not_registered_when_settings_are_disabled(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
 
-    def test_phone_methods_are_allowed_when_user_belongs_to_messaging_domain(self):
+    def test_phone_methods_are_registered_when_user_belongs_to_messaging_domain(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertTrue(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator', 'sms', 'call'], [m.code for m in registry.get_methods()])
 
-    def test_phone_methods_are_prohibited_when_user_does_not_belongs_to_messaging_domain(self):
+    def test_phone_methods_are_not_registered_when_user_does_not_belongs_to_messaging_domain(self):
         user = self._create_user(belongs_to_messaging_domain=False)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
+
+    def test_custom_methods_are_registsered(self):
+        user = self._create_user(belongs_to_messaging_domain=True)
+        self._create_view_for_user(user)
+
+        generator_method = registry.get_method('generator')
+        self.assertEqual(HQGeneratorMethod, type(generator_method))
```

Comment:
I wonder if there's a more intuitive way to test this. I'm guessing our previous code required setting up a view to verify the 2FA process, but it seems the new API has changed that where we'd only need to call something like `setup_2fa_registry()` before making these verifications. Would that make sense to configure the setup process that way?

Reply:
This was removed in 4ccbb9f9c74ee16ae5179cd22299b53e1dbdcabf. I think the tests now operate similar to how you've specified, asserting that after calling `setup_2fa_registry`, the values are as expected.